app/controllers/api/stateless/go_cardless/payment_methods_controller.rb

```diff
@@ -8,6 +8,19 @@ class PaymentMethodsController < StatelessController
         def index
           @payment_methods = PaymentHelper::GoCardless.payment_methods_for_member(@current_member)
         end
+
+        def destroy
+          @payment_method = PaymentHelper::GoCardless.payment_method_for_member(member: @current_member, id: params[:id])
+          PaymentProcessor::GoCardless::Populator.client.mandates.cancel(@payment_method.go_cardless_id)
+          @payment_method.update(cancelled_at: Time.now)
+          render json: { success: true }
+        rescue GoCardlessPro::InvalidApiUsageError,
+          GoCardlessPro::InvalidStateError,
```

Comment:
@osahyoun I've been explicit on the type now. I'm rescuing four different GC errors. I don't want to use the same service for both mandates and subscriptions because in the error message it should be clear which cancellation is failing. This is essentially seven lines of code (more now due to listing all the different errors), and I understand wanting to keep controllers slim, but extracting it into a service class with much more lines feels like an overkill.